src/sql/workbench/services/commandLine/common/commandLineService.ts

```diff
@@ -34,52 +32,62 @@ export class CommandLineService implements ICommandLineProcessing {
 		@IWorkspaceConfigurationService private _configurationService: IWorkspaceConfigurationService
 	) {
-		let profile = null;
-		if (this._environmentService) {
-			if (this._commandService) {
-				this._commandName = this._environmentService.args.command;
-			}
-			if (this._environmentService.args.server) {
-				profile = new ConnectionProfile(_capabilitiesService, null);
-				// We want connection store to use any matching password it finds
-				profile.savePassword = true;
-				profile.providerName = Constants.mssqlProviderName;
-				profile.serverName = _environmentService.args.server;
-				profile.databaseName = _environmentService.args.database ? _environmentService.args.database : '';
-				profile.userName = _environmentService.args.user ? _environmentService.args.user : '';
-				profile.authenticationType = _environmentService.args.integrated ? 'Integrated' : 'SqlLogin';
-				profile.connectionName = '';
-				profile.setOptionValue('applicationName', Constants.applicationName);
-				profile.setOptionValue('databaseDisplayName', profile.databaseName);
-				profile.setOptionValue('groupId', profile.groupId);
-			}
+		if (ipc) {
+		    ipc.on('ads:processCommandLine', (event: any, args: ParsedArgs) => this.onLaunched(args));
 		}
-		this._connectionProfile = profile;
+		// we only get the ipc from main during window reuse
+		this.onLaunched(_environmentService.args);
+	}
+
+	private onLaunched(args: ParsedArgs)
+	{
 		const registry = platform.Registry.as<IConnectionProviderRegistry>(ConnectionProviderExtensions.ConnectionProviderContributions);
 		let sqlProvider = registry.getProperties(Constants.mssqlProviderName);
 		// We can't connect to object explorer until the MSSQL connection provider is registered
 		if (sqlProvider) {
-			this.processCommandLine().catch(reason => { warn('processCommandLine failed: ' + reason); });
+			this.processCommandLine(args).catch(reason => { warn('processCommandLine failed: ' + reason); });
 		} else {
 			registry.onNewProvider(e => {
 				if (e.id === Constants.mssqlProviderName) {
-					this.processCommandLine().catch(reason => { warn('processCommandLine failed: ' + reason); });
+					this.processCommandLine(args).catch(reason => { warn('processCommandLine failed: ' + reason); });
 				}
 			});
 		}
 	}
+
 	public _serviceBrand: any;
 	// We base our logic on the combination of (server, command) values.
 	// (serverName, commandName) => Connect object explorer and execute the command, passing the connection profile to the command. Do not load query editor.
 	// (null, commandName) => Launch the command with a null connection. If the command implementation needs a connection, it will need to create it.
 	// (serverName, null) => Connect object explorer and open a new query editor
 	// (null, null) => Prompt for a connection unless there are registered servers
-	public processCommandLine(): Promise<void> {
+	public processCommandLine(args: ParsedArgs): Promise<void> {
+		let profile = null;
```

Comment:
we should use `undefined` instead of `null` to be consistent with project conventions.